modules/unpickler/src/main/scala/ch/epfl/scala/debugadapter/internal/stacktrace/Scala3Unpickler.scala

```diff
@@ -75,33 +72,41 @@ class Scala3Unpickler(
               term <- collectLocalSymbols(owner) {
                 case (t: TermSymbol, None) if (t.isLazyVal || t.isModuleVal) && t.matchName(name) => t
               }
-            yield term
+            yield BinaryMethod(bcls, term, BinaryMethodKind.LocalLazyInit)
           case AnonFun(prefix) =>
-            val x =
-              for
-                owner <- withCompanionIfExtendsAnyVal(cls)
-                term <- collectLocalSymbols(owner) {
-                  case (t: TermSymbol, None) if t.isAnonFun && matchSignature(method, t) => t
-                }
-              yield term
-            if x.size > 1 && prefix.nonEmpty then
-              val y = x.filter(s => matchPrefix(prefix, s.owner))
-              if y.size == 0 then x
-              else y
-            else x
+            val symbols =
+              val x =
+                for
+                  owner <- withCompanionIfExtendsAnyVal(cls)
+                  term <- collectLocalSymbols(owner) {
+                    case (t: TermSymbol, None) if t.isAnonFun && matchSignature(method, t) => t
+                  }
+                yield term
+              if x.size > 1 && prefix.nonEmpty then
+                val y = x.filter(s => matchPrefix(prefix, s.owner))
+                if y.size == 0 then x
+                else y
+              else x
+            symbols.map(term => BinaryMethod(bcls, term, BinaryMethodKind.AnonFun))
           case LocalMethod(name, _) =>
             for
               owner <- withCompanionIfExtendsAnyVal(cls)
               term <- collectLocalSymbols(owner) {
                 case (t: TermSymbol, None) if t.matchName(name) && matchSignature(method, t) => t
               }
-            yield term
+            yield BinaryMethod(bcls, term, BinaryMethodKind.LocalDef)
           case LazyInit(name) =>
-            cls.declarations.collect { case t: TermSymbol if t.isLazyVal && t.matchName(name) => t }
+            cls.declarations.collect {
+              case t: TermSymbol if t.isLazyVal && t.matchName(name) => BinaryMethod(bcls, t, BinaryMethodKind.LazyInit)
+            }
           case _ =>
             cls.declarations
-              .collect { case sym: TermSymbol => sym }
-              .filter(matchSymbol(method, _))
+              .collect { case sym: TermSymbol =>
+                if method.name == "$init$" then BinaryMethod(bcls, sym, BinaryMethodKind.Constructor)
+                else if method.name == "<init>" then BinaryMethod(bcls, sym, BinaryMethodKind.TraitConstructor)
```

Comment:
I think it's the opposite:

```suggestion
                if method.name == "$init$" then BinaryMethod(bcls, sym, BinaryMethodKind.TraitConstructor)
                else if method.name == "<init>" then BinaryMethod(bcls, sym, BinaryMethodKind.Constructor)
```

Maybe you should add a new assert method in the tests to check the kind.